### ISSUE TITLE
Unable to install custom component with attached zip assets

### ISSUE THREAD

**Issue by mfabiani53** [hacs/integration#922]
## Version of HACS
0.20.7

## Describe the bug

I've inspected the issue and it seems that such thing happens if the release you are trying to install via HACS has a third attached ZIP file more than the sources. In particular, I wanted attach a zip file containing only **/custom_components/****, named "meross_cloud.zip". If you do that, HACS fails to download the source zip file with the error "download was incomplete". After I've removed the third asset, everything went back to normal (i.e. working).

***

I am still unable to update/install the latest release of this integration. First of all i tried to update and got the same errors, then i uninstalled and tried to reinstall unsuccesfully.
I installed manually the integration, but in this way i will not be notified about new releases.
See the log for more infos..

## Debug log

```
2020-01-08 12:07:33 DEBUG (MainThread) [hacs] 4537
2020-01-08 12:07:33 DEBUG (MainThread) [hacs] 2020-01-08 12:05:04
2020-01-08 12:07:37 INFO (MainThread) [hacs] Loading known repositories
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.factory] No tasks to execute
2020-01-08 12:08:01 DEBUG (MainThread) [hacs] Starting recuring background task for installed repositories
2020-01-08 12:08:01 DEBUG (MainThread) [hacs] 4524
2020-01-08 12:08:01 DEBUG (MainThread) [hacs] 2020-01-08 12:05:04
2020-01-08 12:08:01 INFO (MainThread) [hacs.factory] Processing 61 tasks
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.integration.hacs.integration] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.integration.custom-components.places] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.integration.custom-components.alexa_media_player] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.integration.custom-components.sensor.rpi_power] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.circle-sensor-card] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.state-attribute-element] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.vertical-stack-in-card] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.dual-gauge-card] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-slider-entity-row] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.username-element] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.integration.custom-components.feedparser] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.list-card] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.button-card] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.upcoming-media-card] Getting repository information
2020-01-08 12:08:01 DEBUG (MainThread) [hacs.repository.integration.custom-components.sensor.sonarr_upcoming_media] Getting repository information
2020-01-08 12:08:16 DEBUG (MainThread) [hacs.repository.integration.custom-components.sensor.radarr_upcoming_media] Getting repository information
2020-01-08 12:08:16 DEBUG (MainThread) [hacs.repository.plugin.kalkih.mini-media-player] Getting repository information
2020-01-08 12:08:16 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-fold-entity-row] Getting repository information
2020-01-08 12:08:16 DEBUG (MainThread) [hacs.repository.plugin.kalkih.mini-graph-card] Getting repository information
2020-01-08 12:08:16 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.light-entity-row] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.button-entity-row] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-layout-card] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.integration.thomasloven.hass-fontawesome] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.nervetattoo.simple-thermostat] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-card-tools] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.bar-card] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-auto-entities] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.atomic7777.atomic_calendar] Getting repository information
2020-01-08 12:08:17 DEBUG (MainThread) [hacs.repository.plugin.kalkih.simple-weather-card] Getting repository information
2020-01-08 12:08:18 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.spotify-card] Getting repository information
2020-01-08 12:08:26 DEBUG (MainThread) [hacs.repository.plugin.gadgetchnnel.lovelace-home-feed-card] Getting repository information
2020-01-08 12:08:26 DEBUG (MainThread) [hacs.repository.plugin.ljmerza.reddit-card] Getting repository information
2020-01-08 12:08:26 DEBUG (MainThread) [hacs.repository.plugin.mishaaq.sun-card] Getting repository information
2020-01-08 12:08:26 DEBUG (MainThread) [hacs.repository.integration.custom-components.gpodder] Getting repository information
2020-01-08 12:08:27 DEBUG (MainThread) [hacs.repository.plugin.ljmerza.travel-time-card] Getting repository information
2020-01-08 12:08:27 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-more-info-card] Getting repository information
2020-01-08 12:08:27 DEBUG (MainThread) [hacs.repository.integration.fondberg.spotcast] Getting repository information
2020-01-08 12:08:27 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.bignumber-card] Getting repository information
2020-01-08 12:08:28 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.entity-attributes-card] Getting repository information
2020-01-08 12:08:28 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.gauge-card] Getting repository information
2020-01-08 12:08:28 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.group-card] Getting repository information
2020-01-08 12:08:28 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.home-setter] Getting repository information
2020-01-08 12:08:28 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.plan-coordinates] Getting repository information
2020-01-08 12:08:30 DEBUG (MainThread) [hacs.repository.plugin.custom-cards.decluttering-card] Getting repository information
2020-01-08 12:08:30 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-markdown-mod] Getting repository information
2020-01-08 12:08:36 DEBUG (MainThread) [hacs.repository.plugin.thomasloven.lovelace-card-mod] Getting repository information
2020-01-08 12:08:36 DEBUG (MainThread) [hacs.repository.plugin.bramkragten.weather-card] Getting repository information
2020-01-08 12:08:36 DEBUG (MainThread) [hacs.repository.integration.thomasloven.hass-browser_mod] Getting repository information
2020-01-08 12:08:37 DEBUG (MainThread) [hacs.repository.integration.snarky-snark.home-assistant-variables] Getting repository information
2020-01-08 12:08:37 DEBUG (MainThread) [hacs.repository.integration.caiosweet.Home-Assistant-custom-components-DPC-Alert] Getting repository information
2020-01-08 12:08:38 DEBUG (MainThread) [hacs.repository.theme.home-assistant-community-themes.amoled] Getting repository information
2020-01-08 12:08:38 DEBUG (MainThread) [hacs.repository.theme.home-assistant-community-themes.dark-mint] Getting repository information
2020-01-08 12:08:38 DEBUG (MainThread) [hacs.repository.plugin.r-renato.ha-card-waze-travel-time] Getting repository information
2020-01-08 12:08:39 DEBUG (MainThread) [hacs.repository.theme.home-assistant-community-themes.blue-night] Getting repository information
2020-01-08 12:08:39 DEBUG (MainThread) [hacs.repository.plugin.bokub.rgb-light-card] Getting repository information
2020-01-08 12:08:39 DEBUG (MainThread) [hacs.repository.theme.home-assistant-community-themes.christmas] Getting repository information
2020-01-08 12:08:41 DEBUG (MainThread) [hacs.template] 'state' is undefined
2020-01-08 12:08:41 DEBUG (MainThread) [hacs.repository.theme.home-assistant-community-themes.material-dark-green] Getting repository information
2020-01-08 12:08:41 DEBUG (MainThread) [hacs.repository.integration.elad-bar.ha-hpprinter] Getting repository information
2020-01-08 12:08:41 DEBUG (MainThread) [hacs.repository.theme.Poeschl.slate_red] Getting repository information
2020-01-08 12:08:41 DEBUG (MainThread) [hacs.repository.theme.arsaboo.oxford_blue_theme] Getting repository information
2020-01-08 12:08:46 DEBUG (MainThread) [hacs.repository.theme.aFFekopp.dark_teal] Getting repository information
2020-01-08 12:08:54 INFO (MainThread) [hacs.factory] Task processing of 61 tasks completed in 53 seconds
2020-01-08 12:08:55 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:08:55 DEBUG (MainThread) [hacs] Recuring background task for installed repositories done
2020-01-08 12:08:55 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:09:38 DEBUG (MainThread) [hacs] Running set_state for albertogeniola/meross-homeassistant
2020-01-08 12:09:38 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:09:38 DEBUG (MainThread) [hacs] Running update for albertogeniola/meross-homeassistant
2020-01-08 12:09:38 DEBUG (MainThread) [hacs.repository.integration.albertogeniola.meross-homeassistant] Getting repository information
2020-01-08 12:09:43 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:09:44 DEBUG (MainThread) [hacs] Running set_state for albertogeniola/meross-homeassistant
2020-01-08 12:09:44 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:09:44 DEBUG (MainThread) [hacs] Running set_version for albertogeniola/meross-homeassistant
2020-01-08 12:09:44 DEBUG (MainThread) [hacs.repository.integration.albertogeniola.meross-homeassistant] Getting repository information
2020-01-08 12:09:48 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:09:54 DEBUG (MainThread) [hacs] Running set_state for albertogeniola/meross-homeassistant
2020-01-08 12:09:54 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:09:54 DEBUG (MainThread) [hacs] Running install for albertogeniola/meross-homeassistant
2020-01-08 12:09:54 DEBUG (MainThread) [hacs.repository.integration.albertogeniola.meross-homeassistant] Getting repository information
2020-01-08 12:09:57 DEBUG (MainThread) [hacs.repository.integration.albertogeniola.meross-homeassistant] Downloading meross_cloud.zip
2020-01-08 12:09:58 DEBUG (MainThread) [hacs.download.downloader] Downloading https://github.com/albertogeniola/meross-homeassistant/releases/download/20200105.8/meross_cloud.zip
2020-01-08 12:09:59 ERROR (MainThread) [hacs.repository.integration.albertogeniola.meross-homeassistant] Download was not complete.
2020-01-08 12:09:59 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:10:03 DEBUG (MainThread) [hacs] Running set_state for albertogeniola/meross-homeassistant
2020-01-08 12:10:03 DEBUG (MainThread) [hacs.data] Saving data
2020-01-08 12:10:03 DEBUG (MainThread) [hacs] Running set_version for albertogeniola/meross-homeassistant
2020-01-08 12:10:03 DEBUG (MainThread) [hacs.repository.integration.albertogeniola.meross-homeassistant] Getting repository information
2020-01-08 12:10:09 DEBUG (MainThread) [hacs.data] Saving data
```

**Comment by benleb** [comment]
quick&ugly fix:
first, stop hass completely. then, in your .storage/hacs.repositories file, search for eross to get to the entry for this repo. then replace the value in "selected_tag" with null, like in the screenshot. then restart home assistant. when installing, DO NOT choose which version you want in the HACS UI when you install it afterwards! and magically it works ;)
![image](https://user-images.githubusercontent.com/512997/72012174-bfd45d80-325b-11ea-96c0-0ad0a3e55050.png)

**Comment by albertogeniola** [comment]
@ludeeus 
Hi all, I'am Alberto Geniola, developer and maintainer of the Meross Cloud integration plugin. I've inspected the issue and it seems that such thing happens if the release you are trying to install via HACS has a third attached ZIP file more than the sources. In particular, I wanted attach a zip file containing only **/custom_components/\*\***, named "meross_cloud.zip". If you do that, HACS fails to download the source zip file with the error "download was incomplete". After I've removed the third asset, everything went back to normal (i.e. working).

I was unable to find any reference in the documentation that forbids to add any asset to a given release. What am I missing?

**Comment by ludeeus** [comment]
It's a bug, will be fixed in a few weeks (needs redesign of the repository objects).

**Comment by alandtse** [comment]
Also ran into this bug for my [component](https://github.com/custom-components/alexa_media_player/issues/513). @mfabiani53 can you generalize the issue? I'd suggest the following edits. 

Title: 
> Unable to install custom component with attached zip assets

Describe the bug:

> I've inspected the issue and it seems that such thing happens if the release you are trying to install via HACS has a third attached ZIP file more than the sources. In particular, I wanted attach a zip file containing only **/custom_components/****, named "meross_cloud.zip". If you do that, HACS fails to download the source zip file with the error "download was incomplete". After I've removed the third asset, everything went back to normal (i.e. working).

Workaround for Devs:

1. Delete extraneous zip assets
2. If zip asset follows custom_component directory format, add following keys to hacs.json
```json
  "zip_release": true,
  "filename": "asset.zip",
```
Example [PR](https://github.com/custom-components/alexa_media_player/pull/517/files).

**Comment by DavidFW1960** [comment]
@ludeeus same cause here as for the issue I opened regarding lovelace cards (not components)? Is there a work around for a lovelace card or is it the same as the above work around for an integration?

**Comment by ludeeus** [comment]
I _think_ this is fixed with 0.21
